pkg/kapp/cmd/core/deps_factory.go

```diff
@@ -40,8 +46,9 @@ func (f *DepsFactoryImpl) DynamicClient() (dynamic.Interface, error) {
 
 	// copy to avoid mutating the passed-in config
 	cpConfig := rest.CopyConfig(config)
-	// set the warning handler for this client to ignore warnings
-	cpConfig.WarningHandler = rest.NoWarnings{}
+	if !f.Warnings {
```

Comment:
within kapp there is a UI object that handles all interactions with stdout/stderr. when warnings are enabled lets redirect them to ui.PrintLinef (e.g. https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/cmd/app/deploy.go#L190)

Reply:
Sounds good. I will take a look at the example.

Reply:
We would have to write our own custom warning handler if we would want to use ui.PrintLinef, which seems fine to me. The only concern is that we would also have to implement deduplication etc,. on our own (duplicating already existing code in vendor).
I will see if we can somehow use their existing handler which accepts an io.Writer to write the warnings, or we will proceed with our own custom warning handler.